src/Orleans.Core/Timers/TimerManager.cs

```diff
@@ -8,40 +8,45 @@ namespace Orleans.Timers.Internal
 {
     public interface ITimerManager
     {
-        Task Delay(TimeSpan timeSpan);
+        Task<bool> Delay(TimeSpan timeSpan, CancellationToken cancellationToken = default(CancellationToken));
     }
 
     internal class TimerManagerImpl : ITimerManager
     {
-        public Task Delay(TimeSpan timeSpan) => TimerManager.Delay(timeSpan);
+        public Task<bool> Delay(TimeSpan timeSpan, CancellationToken cancellationToken = default(CancellationToken)) => TimerManager.Delay(timeSpan, cancellationToken);
     }
 
     internal static class TimerManager
     {
-        public static Task Delay(TimeSpan timeSpan) => DelayUntil(DateTime.UtcNow + timeSpan);
+        public static Task<bool> Delay(TimeSpan timeSpan, CancellationToken cancellationToken = default(CancellationToken)) => DelayUntil(DateTime.UtcNow + timeSpan, cancellationToken);
 
-        public static Task DelayUntil(DateTime dueTime)
+        public static Task<bool> DelayUntil(DateTime dueTime, CancellationToken cancellationToken = default(CancellationToken))
         {
-            var result = new DelayTimer(dueTime);
+            var result = new DelayTimer(dueTime, cancellationToken);
             TimerManager<DelayTimer>.Register(result);
             return result.Completion;
         }
 
         private sealed class DelayTimer : ITimerCallback, ILinkedListElement<DelayTimer>
         {
-            private readonly TaskCompletionSource<int> completion =
-                new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
+            private readonly TaskCompletionSource<bool> completion =
+                new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
 
-            public DelayTimer(DateTime dueTime)
+            public DelayTimer(DateTime dueTime, CancellationToken cancellationToken)
             {
                 this.DueTime = dueTime;
+                this.CancellationToken = cancellationToken;
             }
 
-            public Task Completion => this.completion.Task;
+            public Task<bool> Completion => this.completion.Task;
```

Comment:
Why return bool?

Reply:
I could revert this. The reasoning was for the caller to be able to distinguish between a canceled and completed delay call without throwing a cancellation exception (cancellation is not exceptional)

Reply:
I'm not opposed to it, just thought since these were intended to replace task.Delay, they should probably follow the same pattern.  I'll leave it to you're discretion.